src/main.cpp

```diff
@@ -90,6 +90,25 @@ void load_name_file(const char* file_path)
 static std::bitset<kernel_api_tests_size> tests_to_run;
 static std::bitset<kernel_api_tests_size> tests_to_exclude;
 
+unsigned long convert_test_api_input(char* test_str) {
+    // Trim space(s) before attempting to compare strings.
+    while(test_str[0] == ' ') {
+        test_str++;
+    }
+    if (strlen(test_str) > 4) {
```

Comment:
Instead of checking for an arbitrary string length, you can just attempt to call directly `strtoul` on the parsed string. If the test is specified with an API name, then the function will fail and will return either 0 or `ULONG_MAX`, and thus you can distinguish between the two cases.

Reply:
Hmm... when I made a sample test of:
```cpp
#include <climits>
#include <iostream>
#include <string>
 
int main()
{
    unsigned long value1 = strtoul("AB", NULL, 16);
    unsigned long value2 = strtoul("0xF", NULL, 16);
    unsigned long value3 = strtoul("10", NULL, 16);
    unsigned long value4 = strtoul("bsomething", NULL, 16);
    unsigned long value5 = strtoul("somethingb", NULL, 16);
    std::cout << "ULONG_MAX: " << ULONG_MAX << "\n";
    std::cout << "value1: " << value1 << "\n";
    std::cout << "value2: " << value2 << "\n";
    std::cout << "value3: " << value3 << "\n";
    std::cout << "value4: " << value4 << "\n";
    std::cout << "value5: " << value5 << "\n";
}
```
The output will return as:
```
ULONG_MAX: 18446744073709551615
value1: 171
value2: 15
value3: 16
value4: 11
value5: 0
```
which make `strtoul` function not entirely reliable. Any valid decimal/hexadecimal characters will be read and converted. Any invalid characters will return what's been found at beginning of the string.

The reason I have use `strlen` over 4 character lengths is normally known used for api name string. Below the length is known for hexadecimal (without `0x` prefix of course). If there's alternative method not to rely on `strtoul` return value, then I'm all for it.

As for `return kernel_api_tests_size;` which indeed should have use `ULONG_MAX` return.

Reply:
Also, smallest API name length so far is 6, `XcHMAC` and `RtlRip`.